src/webgpu/api/validation/capability_checks/features/texture_component_swizzle.spec.ts

```diff
@@ -0,0 +1,231 @@
+export const description = `
+Validation tests for the 'texture-component-swizzle' feature.
+
+Test that:
+* when the feature is off, swizzling is not allowed, even the identity swizzle.
+* swizzling is not allowed on textures with usage STORAGE_BINDING nor RENDER_ATTACHMENT
+  except the identity swizzle.
+`;
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { GPUConst } from '../../../../constants.js';
+import { UniqueFeaturesOrLimitsGPUTest } from '../../../../gpu_test.js';
+
+import { isIdentitySwizzle, kSwizzleTests } from './texture_component_swizzle_utils.js';
+
+export const g = makeTestGroup(UniqueFeaturesOrLimitsGPUTest);
+
+g.test('invalid_swizzle')
+  .desc(
+    `
+  Test that setting an invalid swizzle value on a texture view throws an exception.
+  `
+  )
+  .params(u =>
+    u.beginSubcases().combine('invalidSwizzle', ['rgbA', 'rgb', 'rgba01', '', 1234, null])
```

Comment:
As highlighted in https://chromium-review.googlesource.com/c/chromium/src/+/6958569/comment/47daa61a_23127924/, we should also test potential lossy cast cases such as `ɲ` (0x0272) and `r` (0x72)